infoschema/tables.go

```diff
@@ -847,6 +848,103 @@ func dataForTableConstraints(schemas []*model.DBInfo) [][]types.Datum {
 	return rows
 }
 
+func dataForKeyColumnUsage(schemas []*model.DBInfo) [][]types.Datum {
+	rows := [][]types.Datum{}
+	for _, schema := range schemas {
+		for _, table := range schema.Tables {
+			rs := keyColumnUsageInTable(schema, table)
+			for _, r := range rs {
+				rows = append(rows, r)
+			}
+		}
+	}
+	return rows
+}
+
+func keyColumnUsageInTable(schema *model.DBInfo, table *model.TableInfo) [][]types.Datum {
+	rows := [][]types.Datum{}
+	if table.PKIsHandle {
+		for i, col := range table.Columns {
+			if mysql.HasPriKeyFlag(col.Flag) {
+				record := types.MakeDatums(
+					catalogVal,        // CONSTRAINT_CATALOG
+					schema.Name.O,     // CONSTRAINT_SCHEMA
+					primaryConstraint, // CONSTRAINT_NAME
+					catalogVal,        // TABLE_CATALOG
+					schema.Name.O,     // TABLE_SCHEMA
+					table.Name.O,      // TABLE_NAME
+					col.Name.O,        // COLUMN_NAME
+					i,                 // ORDINAL_POSITION
+					1,                 // POSITION_IN_UNIQUE_CONSTRAINT
+					nil,               // REFERENCED_TABLE_SCHEMA
+					nil,               // REFERENCED_TABLE_NAME
+					nil,               // REFERENCED_COLUMN_NAME
+				)
+				rows = append(rows, record)
+				break
+			}
+		}
+	}
+	nameToCol := make(map[string]*model.ColumnInfo, len(table.Columns))
+	for _, c := range table.Columns {
+		nameToCol[c.Name.L] = c
+	}
+	for _, index := range table.Indices {
+		var idxName string
+		if index.Primary {
+			idxName = primaryConstraint
+		} else if index.Unique {
+			idxName = index.Name.O
+		} else {
+			// Only handle unique/primary key
+			continue
+		}
+		for i, key := range index.Columns {
+			col := nameToCol[key.Name.L]
+			record := types.MakeDatums(
+				catalogVal,    // CONSTRAINT_CATALOG
+				schema.Name.O, // CONSTRAINT_SCHEMA
+				idxName,       // CONSTRAINT_NAME
+				catalogVal,    // TABLE_CATALOG
+				schema.Name.O, // TABLE_SCHEMA
+				table.Name.O,  // TABLE_NAME
+				col.Name.O,    // COLUMN_NAME
+				i+1,           // ORDINAL_POSITION,
```

Comment:
Line 877 should be 1.
The value of ORDINAL_POSITION is the column's position within the constraint, not the column's position within the table. Column positions are numbered beginning with 1.
So for PkAsHandle, it should always be 1.